### DISCUSSION TITLE
Make CanvasGroup node that's compatible with Controls

### DISCUSSION THREAD

**Discussion by markdibarry** [godotengine/godot-proposals#4392] [GUI]
`Node2D`s aren't the only ones that suffer from the individual draw call opacity issue. Is there a reason why `CanvasGroup` node is `Node2D` only? If having two separate versions is too tedious, is there a reason why this can't be implemented as part of CanvasItem?

**Comment by phnix-dev** [comment]
Also, the CanvasGroup node works with Control children. (e.g. with a TextureRect node)
Does anyone know how to implement the shader code from the CanvasGroup [documentation](https://docs.godotengine.org/en/stable/classes/class_canvasgroup.html) on a Control node?

**Comment by Megalukes** [comment]
Any updates or solutions about this? You can actually use canvas groups but this will bug the size/position inheritance from parent control nodes
![image](https://github.com/user-attachments/assets/cf19d28d-b5ce-4964-b67b-5e86837aab99)
If I do something like the image above, it works perfectly when it comes to opacity, but it doesn't respect BaseContainer's size/position constraints. Any help would be appreciate it!

**Reply by phnix-dev** [reply]
Unfortunately, you might have to recreate the behavior with shader code or modify the engine to make a copy of the canvas group class and inherit it to the control node (this would mean creating a new c++ container node using the canvas group code, but I do not know if it can work).

**Reply by tsutsen** [reply]
I've made a GDExtension to solve this issue: https://github.com/tsutsen/CanvasGroupContainer. It adds a `CanvasGroupContainer` node that renders like a `CanvasGroup` but iherits all the nice layout feature from `Control`. And here's the proposal: https://github.com/godotengine/godot-proposals/issues/13338

**Reply by Mickeon** [reply]
For the sake of linking, here's https://github.com/godotengine/godot/pull/111380

**Comment by gusg21** [comment]
Would also be great to have mouse_entered and mouse_exited events aggregated for CanvasGroups on Controls, useful for buttons over buttons, etc.!
